app/controllers/concerns/accountable.rb

```diff
@@ -14,8 +14,27 @@ def create_user_account
     log e
   end
 
+  def update_account_login_stats
+    return unless login_stats.present? && login_type.in?(AccountLoginStat::LOGIN_TYPES)
+
+    login_stats.update("#{login_type}_at" => Time.zone.now)
+  end
+
   private
 
+  def login_stats
+    @login_stats ||=
+      if @current_user.account.blank?
+        nil
+      else
+        AccountLoginStat.find_or_initialize_by(account_id: @current_user.account.id)
```

Comment:
You could invert the if condition's logic to `if @current_user.account.present?` and then you wouldn't need an else block.